src/cooperative_IDM.jl

```diff
@@ -13,18 +13,12 @@ the car follows the IntelligentDriverModel.
     - `env::MergingEnvironment = MergingEnvironment(main_lane_angle = 0.0, merge_lane_angle = pi/6)` the merging environment
     - `idm::IntelligentDriverModel = IntelligentDriverModel(v_des = env.main_lane_vmax, d_cmf = 2.0, d_max=2.0, T = 1.5, s_min = 2.0, a_max = 2.0)` the default IDM
     - `c::Float64 = 0.0` the cooperation level
-    - `fov::Float64 = 20.0` [m] A field of view, the merging vehicle is not considered if it is further than `fov`
 """
 @with_kw mutable struct CooperativeIDM <: DriverModel{LaneFollowingAccel}
```

Comment:
Should this belon to the main ADM repo?

Reply:
I have significantly modified it from your version because there wasn't a clear main lane, merge lane distinction, and also there were multiple vehicles on what you called merge lane as opposed to single vehicle. Also, as it is now, it is specific to this roadway by hardcoding the lanes where merging takes place. So, I am wondering how to stress test it enough to make it a part of ADM (or AutomotiveSimulator.jl after name change)